src/components/ArticlesByTopic.jsx

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from "react";
+import { fetchArticles } from "../api";
+import { Link, useLocation } from "react-router-dom";
+import ArticleCard from "./ArticleCard";
+
+export default function ArticlesByTopic() {
+  const topic = useLocation().state;
```

Comment:
You shouldn't need this as you can access the topic from the parametric topic in the url